winapps/src/quickemu.rs

```
@@ -0,0 +1,88 @@
+use crate::{get_data_dir, save_config, Config};
+use std::fs;
+use std::process::exit;
+use std::process::Command;
+
+pub fn create_vm(mut config: Config) {
+    let data_dir = get_data_dir();
+
+    let output = match Command::new("quickget")
+        .current_dir(data_dir)
+        .arg("windows")
+        .arg("10")
+        .output()
+    {
+        Ok(o) => o,
+        Err(e) => {
+            println!("Failed to execute quickget: {}", e);
+            println!("Please make sure quickget is installed and in your PATH");
+            exit(1);
+        }
+    };
+
+    config.vm.name = "windows-10-22H2".to_string();
+    config.vm.short_name = "windows-10".to_string();
+
+    save_config(&config, None).expect("Failed to save config, VM will not start properly");
+
+    println!("{}", String::from_utf8_lossy(&output.stdout));
+}
+
+pub fn start_vm(config: Config) {
+    let data_dir = get_data_dir();
+
+    let command = match Command::new("quickemu")
+        .current_dir(data_dir)
+        .args([
+            "--vm",
+            &format!("{}.conf", config.vm.name),
+            "--display",
+            "none",
```

Is it possible to write --display none in one line (just for better readability)

---

Also @AkechiShiro noticed a option about ignoring msrs per default

---

I don't think we should ignore msrs by default, there's a reason this warning shows up
I think we should at least warn the user about it

---

> Is it possible to write --display none in one line (just for better readability)

Only one argument can be passed per element in the list and I can't put both elements on one line inside the list because rustfmt doesn't like that

---

> I don't think we should ignore msrs by default, there's a reason this warning shows up
> I think we should at least warn the user about it

As far as I know msrs should be disabled on macos and windows guest. Because it prints following message:

echo " - MSR:      WARNING! Ignoring unhandled Model-Specific Registers is disabled."
echo
echo "             echo 1 | sudo tee /sys/module/kvm/parameters/ignore_msrs"
echo
echo "             If you are unable to run macOS or Windows VMs then run the above 👆"
echo "             This will enable ignoring of unhandled MSRs until you reboot the host."
echo "             You can make this change permenant by running: 'quickemu --ignore-msrs-always'"

It seems like this could cause problems on mac and windows vms.

---

And it warns that msrs is NOT ignored.

---

> > Is it possible to write --display none in one line (just for better readability)
> 
> Only one argument can be passed per element in the list and I can't put both elements on one line inside the list because rustfmt doesn't like that

I meant putting it in the same string like:
"--display none"

---

That's not supported, like I said

---

> And it warns that msrs is NOT ignored.

Right, I thought the flag would just ignore the error message and not msrs...
I'll add it